course/include/isometry.h

```diff
@@ -92,21 +93,52 @@ class Matrix3 {
     Matrix3 & operator /= (const double & obj);
     Matrix3 operator * (const double & obj) const {return Matrix3(*this) *= obj;};
     Matrix3 operator / (const double & obj) const {return Matrix3(*this) /= obj;};
-    Vector3 operator * (const Vector3 & obj);
+    Vector3 operator * (const Vector3 & obj) const;
     Vector3 operator [] (const int index) const;
     Vector3 & operator [] (const int index);
     double det() const;
     friend std::ostream& operator << (std::ostream & os, const Matrix3 & v) {
-              os << std::string("[[") << v[0][0] << ", " << v[0][1] << ", " << v[0][2] << "], [";
-              os << v[1][0] << ", " << v[1][1] << ", " << v[1][2] << "], [";
-              os << v[2][0] << ", " << v[2][1] << ", " << v[2][2] << "]]";
+              os << std::string("[[") << std::setprecision(9) << v[0][0] << ", " << std::setprecision(9) << v[0][1] << ", " << std::setprecision(9) << v[0][2] << "], [";
+              os << std::setprecision(9) << v[1][0] << ", " << std::setprecision(9) << v[1][1] << ", " << std::setprecision(9) << v[1][2] << "], [";
+              os << std::setprecision(9) << v[2][0] << ", " << std::setprecision(9) << v[2][1] << ", " << std::setprecision(9)<< v[2][2] << "]]";
               return os;
-          }; //TODO INCREASE NUMBERS TO 4 DECIMALS
+          };
   private:
     Vector3 row1_, row2_, row3_;
 };
 
 Matrix3 operator * (const double & obj1, const Matrix3 & obj2);
 
+class Isometry {
+  public:
+    Isometry(std::initializer_list<double> point, const Matrix3 & rotation): point_(point), rotation_(rotation) {};
+    Isometry(const Vector3 & point, const Matrix3 & rotation): point_(point), rotation_(rotation) {};
+    Isometry(const Isometry & obj): point_(obj.point()), rotation_(obj.rotation()) {};
+    static Isometry FromTranslation(std::initializer_list<double> point){return Isometry(point, Matrix3::kIdentity);};
+    static Isometry RotateAround(const Vector3 & point, const double & angle);
+    static Isometry FromEulerAngles(const double & roll, const double & pitch, const double & yaw);
+
+    const Vector3 & point() const {return point_;};
+    const Matrix3 & rotation() const {return rotation_;};
+    const Vector3 & translation() const {return point_;};
+
+    bool operator == (const Isometry & obj) const;
+    Vector3 operator *= (const Vector3 & obj) const;
```

Comment:
With-assignment operators should always return the same type as the object being updated. Remove this overload, the operation it describes is not really meaningful.